### PR TITLE
#575 One click email unsubscribe

### PR DESCRIPTION
**Pull request by oshiho3**
Implementation for #575. Please review.

### CODE REVIEW

File: app/controllers/members_controller.rb [hunk 1]
```diff
@@ -49,6 +49,27 @@ def view_followers
...
+
+  def unsubscribe
+    begin
+      verifier = ActiveSupport::MessageVerifier.new(ENV['RAILS_SECRET_TOKEN'])
+      decrypted_message = verifier.verify(params[:message])
+
+      @member = Member.find(decrypted_message[:member_id])
+      @type = decrypted_message[:type]
+      @member.update_attributes(@type => false)
+
```
Comment (tygriffin) [inline app/controllers/members_controller.rb]: *Nitpicky note*: Instead of `update_attributes` we can just use `update`.

**Comment by tygriffin** [comment]
Nice! I feel like I'm learning so much from your PR's these days :)

File: app/controllers/members_controller.rb (hunk 1 above)
Comment (oshiho3) [inline app/controllers/members_controller.rb]: good suggestion! Just learned <code>update_attribetes</code> is an alias of <code>update</code>!

**Comment by oshiho3** [comment]
Thanks @tygriffin for checking my code! You're very good at in motivating me :)

File: app/views/notifier/notify.html.haml [hunk 2]
```diff
@@ -18,6 +18,7 @@
   %br/
   = link_to "View this message in your inbox", notification_url(@notification)
   %br/
-  = link_to "Turn off these notifications", edit_member_registration_url
+  = link_to "Turn off these notifications", unsubscribe_member_url(@signed_message)
```
Comment (Skud) [inline app/views/notifier/notify.html.haml]: I'm happy to merge this PR but I would like this string changed to "Unsubscribe from these notifications" (with the word "Unsubscribe" linked) as I think that's what people scan for when looking for a link to click.

Comment (Skud) [inline app/views/notifier/notify.html.haml]: Or even better, "Unsubscribe from direct message notifications" for this one, "... from planting reminders" for planting reminders, etc.  So it's clear you're unsubscribing from one particular kind of message.

**Comment by oshiho3** [comment]
Hi @Skud, I was hoping someone's gonna say that ;-)
I modified the wording accordingly.

**Comment by cesy** [comment]
This carries the security risk that if you forward an email, then someone can unsubscribe you, right? I think that's okay, though. Can I get a second opinion from @pozorvlak or @maco before merging? This looks good to me, though.

**Comment by maco** [comment]
Yes, the forward unsubscribe possibility exists, but it is an extremely
common thing on the web, and as a user I'd rather someone accidentally
unsubscribe me than run into the situation (again! I've hit this with other
sites) of hitting "unsubscribe" and it saying "log in" and me going
"but...I don't...password... ugh, I'll just mark them as spam rather than
deal with figuring out how to log in and unsubscribe."

I believe @pozorvlak already said he agree that the forward unsubscribe
thing being a possibility isn't a problem for merging

**Comment by pozorvlak** [comment]
Yep, looks good to me!

**Pull request merged**